test/com/google/enterprise/adaptor/database/UniqueKeyTest.java

```diff
@@ -104,6 +104,32 @@ public void testNameRepeatsNotAllowed() {
   }
 
   @Test
+  public void testCaseSensitiveNameRepeatsAllowed() {
+    UniqueKey uk = new UniqueKey("num:int,Num:string");
+    assertEquals(2, uk.numElementsForTest());
+    assertEquals("num", uk.nameForTest(0));
+    assertEquals("Num", uk.nameForTest(1));
+    assertEquals(UniqueKey.ColumnType.INT, uk.typeForTest(0));
+    assertEquals(UniqueKey.ColumnType.STRING, uk.typeForTest(1));
+  }
+
+  @Test
+  public void testCaseInsensitiveNameNotAllowedIfCaseSensitiveKeys() {
+    thrown.expect(InvalidConfigurationException.class);
+    UniqueKey uk = new UniqueKey("id:int,Id:string", "ID", "ID");
```

Comment:
A little more complicated than that, since you have to maintain the original map for the types.